monetio/sat/_tropomi_l2_no2_mm.py

```diff
@@ -0,0 +1,138 @@
+"""Read TROPOMI L2 NO2 data."""
+
+import logging
+import os
+import sys
+from collections import OrderedDict
+from glob import glob
+
+import numpy as np
+import xarray as xr
+from netCDF4 import Dataset
+
+
+def _open_one_dataset(fname, variable_dict):
+    """
+    Parameters
+    ----------
+    fname : str
+        Input file path.
+    variable_dict : dict
+
+    Returns
+    -------
+    xarray.Dataset
+    """
+    print("reading " + fname)
+
+    ds = xr.Dataset()
+
+    dso = Dataset(fname, "r")
+
+    longitude = dso.groups["PRODUCT"]["longitude"]
+    latitude = dso.groups["PRODUCT"]["latitude"]
+    start_time = dso.groups["PRODUCT"]["time"]
+
+    # squeeze 1-dimension
+    longitude = np.squeeze(longitude)
+    latitude = np.squeeze(latitude)
+    start_time = np.squeeze(start_time)
+
+    ds["lon"] = xr.DataArray(longitude)
+    ds["lat"] = xr.DataArray(latitude)
+
+    for varname in variable_dict:
+        print(varname)
+        values = dso.groups["PRODUCT"][varname]
+        # squeeze out 1-dimension
+        values = np.squeeze(values)
+
+        if "fillvalue" in variable_dict[varname]:
+            fillvalue = variable_dict[varname]["fillvalue"]
+            values[:][values[:] == fillvalue] = np.nan
+
+        if "scale" in variable_dict[varname]:
+            values[:] = variable_dict[varname]["scale"] * values[:]
+
+        if "minimum" in variable_dict[varname]:
+            minimum = variable_dict[varname]["minimum"]
+            values[:][values[:] < minimum] = np.nan
+
+        if "maximum" in variable_dict[varname]:
+            maximum = variable_dict[varname]["maximum"]
+            values[:][values[:] > maximum] = np.nan
+
+        ds[varname] = xr.DataArray(values)
+
+        if "quality_flag_min" in variable_dict[varname]:
+            ds.attrs["quality_flag"] = varname
+            ds.attrs["quality_thresh_min"] = variable_dict[varname]["quality_flag_min"]
```

Comment:
These are dataset attrs so they will be overwritten if there is more than one variable in `variable_dict`.

Reply:
True. This is an old version of the TROPOMI reader. I add a new dataset attribute 'var_applied' to define the variables list for each quality flag in the new reader

Reply:
@mlirenzhenmayi I don't see this in the current version of the code.